src/core/ir/eval_http.rs

```diff
@@ -109,6 +113,24 @@ impl<'a, 'ctx, Context: ResolverContextLike + Sync> EvalHttp<'a, 'ctx, Context>
                 }
             },
             None => self.execute(request).await,
+        };
+
+        // send the final response to JS script to futher evaluation.
+        if let Ok(resp) = resp {
+            if let Some(on_response) = http_filter.on_response.as_ref() {
+                let js_response = worker::WorkerResponse::try_from(resp.clone())?;
+                let response_event = worker::Event::Response(js_response);
+                let final_command = worker.call(on_response, response_event).await?;
```

Comment:
The worker response should not send the response body to the JS function. We have `on_response_body` for that. Event worker response is going to be a little different than the command worker response.

Reply:
Ignore everything above. I think we don't need any of this code. `on_response_body` should only send the response body to the JS function.